spec/models/users/scopes/having_reminder_mail_to_send_spec.rb

```diff
@@ -0,0 +1,465 @@
+  let(:users) { [paris_user] }
+
+  before do
+    allow(Time).to receive(:current).and_return(current_utc_time)
+    notifications
+    users
+  end
+
+  context 'for a user whose local time is matching the configured time' do
+    it 'contains the user' do
+      expect(scope)
+        .to match_array([paris_user])
+    end
+  end
+
+  context 'for a user whose local time is not matching the configured time' do
+    let(:current_utc_time) { ActiveSupport::TimeZone['UTC'].parse("08:20:59") }
+    let(:scope_time) { ActiveSupport::TimeZone['UTC'].parse("08:15") }
+
+    it 'is empty' do
+      expect(scope)
+        .to be_empty
+    end
+  end
+
+  context 'for a user whose local time is matching the configured time (in a non CET time zone)' do
+    let(:moscow_user) do
+      FactoryBot.create(
+        :user,
+        firstname: 'Moscow',
+        preferences: {
+          time_zone: "Moscow",
+          daily_reminders: {
+            enabled: true,
+            times: [hitting_reminder_slot_for("Moscow", current_utc_time)]
+          }
+        }
+      )
+    end
+    let(:notifications) do
+      FactoryBot.create(:notification, recipient: moscow_user, created_at: 5.minutes.ago)
+    end
+    let(:users) { [moscow_user] }
+
+    it 'contains the user' do
+      expect(scope)
+        .to match_array([moscow_user])
+    end
+  end
+
+  context 'for a user whose local time is matching one of the configured times' do
+    let(:paris_user_daily_reminders) do
+      {
+        enabled: true,
+        times: [
+          hitting_reminder_slot_for("Paris", current_utc_time - 3.hours),
+          hitting_reminder_slot_for("Paris", current_utc_time),
+          hitting_reminder_slot_for("Paris", current_utc_time + 3.hours)
+        ]
+      }
+    end
+
+    it 'contains the user' do
+      expect(scope)
+        .to match_array([paris_user])
+    end
+  end
+
+  context 'for a user who has configured a slot between the earliest_time (in local time) and his current local time' do
+    let(:paris_user_daily_reminders) do
+      {
+        enabled: true,
+        times: [
+          hitting_reminder_slot_for("Paris", current_utc_time - 2.hours),
+          hitting_reminder_slot_for("Paris", current_utc_time + 3.hours)
+        ]
+      }
+    end
+    let(:scope_time) { ActiveSupport::TimeZone['UTC'].parse("06:00") }
+
+    it 'contains the user' do
+      expect(scope)
+        .to match_array([paris_user])
+    end
+  end
+
+  context 'for a user who has configured a slot before the earliest_time (in local time) and after his current local time' do
+    let(:paris_user_daily_reminders) do
+      {
+        enabled: true,
+        times: [
+          hitting_reminder_slot_for("Paris", current_utc_time - 3.hours),
+          hitting_reminder_slot_for("Paris", current_utc_time + 1.hour)
+        ]
+      }
+    end
+    let(:scope_time) { Time.current - 2.hours }
```

Comment:
```suggestion
    let(:scope_time) { current_utc_time - 2.hours }
```
I believe that this would improve readability as you do not need to keep in your head that `Time.current` was mocked.